packages/beacon-node/src/metrics/metrics/lodestar.ts

```diff
@@ -649,27 +649,44 @@ export function createLodestarMetrics(
       receivedToGossipValidate: register.histogram({
         name: "lodestar_gossip_block_received_to_gossip_validate",
         help: "Time elapsed between block received and block validated",
-        buckets: [0.05, 0.1, 0.2, 0.5, 1, 1.5, 2, 4],
+        buckets: [0.05, 0.1, 0.3, 0.5, 0.7, 1, 1.3, 1.6, 2, 2.5, 3, 3.5, 4],
       }),
       receivedToStateTransition: register.histogram({
         name: "lodestar_gossip_block_received_to_state_transition",
         help: "Time elapsed between block received and block state transition",
-        buckets: [0.05, 0.1, 0.2, 0.5, 1, 1.5, 2, 4],
+        buckets: [0.05, 0.1, 0.3, 0.5, 0.7, 1, 1.3, 1.6, 2, 2.5, 3, 3.5, 4],
       }),
       receivedToSignaturesVerification: register.histogram({
         name: "lodestar_gossip_block_received_to_signatures_verification",
         help: "Time elapsed between block received and block signatures verification",
-        buckets: [0.05, 0.1, 0.2, 0.5, 1, 1.5, 2, 4],
+        buckets: [0.05, 0.1, 0.3, 0.5, 0.7, 1, 1.3, 1.6, 2, 2.5, 3, 3.5, 4],
       }),
       receivedToExecutionPayloadVerification: register.histogram({
         name: "lodestar_gossip_block_received_to_execution_payload_verification",
         help: "Time elapsed between block received and execution payload verification",
-        buckets: [0.05, 0.1, 0.2, 0.5, 1, 1.5, 2, 4],
+        buckets: [0.05, 0.1, 0.3, 0.5, 0.7, 1, 1.3, 1.6, 2, 2.5, 3, 3.5, 4],
+      }),
+      receivedToBlobsAvailabilityTime: register.histogram<"numBlobs">({
+        name: "lodestar_gossip_block_received_to_blobs_availability_time",
+        help: "Time elapsed between block received and blobs became available",
+        buckets: [0.05, 0.1, 0.3, 0.5, 0.7, 1, 1.3, 1.6, 2, 2.5, 3, 3.5, 4],
+        labelNames: ["numBlobs"],
```

Comment:
How many different values do we expect `numBlobs` to have? This metric could be really expensive due to the high amount of buckets

Reply:
we need more precise availability time bucketing, i don't think bucketing should affect performance atleast on lodestar side, on grafana side, its very easy to build and query the indices which is a solved problem in relational DBs

Reply:
prom-client has to set a record for each label x value combination, I don't think it is an issue for single label metrics but if you have 2 or more the amount of records created grows pretty fast and since histograms always have a `le` label by default we need to keep that in mind.

There was a performance fix @tuyennhv did on js-libp2p-gossipsub related to that
- https://github.com/ChainSafe/js-libp2p-gossipsub/pull/433

Also keep in mind that Lodestar has to render the metrics as string to be consumed by prometheus, there are plenty of performance issues on prom-client related to that
- https://github.com/siimon/prom-client/pull/549
- https://github.com/siimon/prom-client/pull/548

I honestly can't tell what the limit should be, if we use low cardinality values it might be fine, i.e. if `numBlobs` only has 2-4 different values

Reply:
> https://github.com/ChainSafe/js-libp2p-gossipsub/pull/433

how much performance gain did it give as compared to the total CPU? Also libp2p metrics would have been called all the time however this is max 6*some factor in 12 seconds, i.e. no real juice here to try to optimize it

> if numBlobs only has 2-4 different values
right now it can have 6 values which can grow later but we can check what we need to do that?

Reply:
> max 6*some factor in 12 seconds, i.e. no real juice here to try to optimize it

While this is true keep in mind that those metrics are still stored in-memory and have to be serialized if /metrics endpoint is called.

> no real juice here to try to optimize it

maybe not, just need to be mindful of it when adding metrics


There are a bunch of performance improvements in prom-client related to labels, we should update to latest version

Reply:
true but lets focus on optimizations which really affect the client and show up on the perf runs, i am sure they will bubble up, imo this isn't one of them (i could be wrong)

But yes, good idea to upgrade prom-client to leverage whatever gains are there :+1: